src/vs/workbench/contrib/notebook/browser/services/notebookExecutionStateServiceImpl.ts

```diff
@@ -142,11 +147,25 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 	private _createNotebookCellExecution(notebook: NotebookTextModel, cellHandle: number): CellExecution {
 		const notebookUri = notebook.uri;
 		const exe: CellExecution = this._instantiationService.createInstance(CellExecution, cellHandle, notebook);
+		const cellUri = CellUri.generate(notebookUri, cellHandle);
+		const audioCuesEnabled = this._configurationService.getValue<boolean>(NotebookSetting.audioCuesEnabled);
+
+		let onDidCompleteCallback: (lastRunSuccess: boolean | undefined) => void;
+		if (audioCuesEnabled) {
+			const audioCuesCallback = this._getAudioCuesOnDidCompleteCallback(cellUri);
+			onDidCompleteCallback = (lastRunSuccess) => {
```

Comment:
Feels like checking the setting outside of the callback is making this overly complicated. Checking it inside the callback might be simpler and also make it work if I change the setting while the cell is executing

Reply:
> make it work if I change the setting while the cell is executing
Actually what I wanted to achieve was the exact contrary. I thought that set the audio cue behaviour when the execution was scheduled would have been more consistent. But I see in terms value in terms of user experience for the behaviour you proposed.